ethers-core/src/types/block.rs

```diff
@@ -81,7 +81,7 @@ pub struct Block<TX> {
     pub mix_hash: Option<H256>,
     /// Nonce
     #[cfg(not(feature = "celo"))]
-    pub nonce: Option<U64>,
+    pub nonce: Option<H64>,
```

Comment:
how does H64 serialize, deserialize compared to U64?

H64 is a hash type, and since nonce is an uint, this change doesn't seem necessary,

also H64 lacks common conversion trait impls for primitive types and I'm not sure it even as a `as_u64` function.

Reply:
H64 will encode as a byte array, and not like number so it won't serialize to the most compact representation and it will expect the length to be 16 on deserialization. The reason this change is necessary is because a client of anvil (in my experience here graph-node) may expect the the value to have a length of 16 (or 18 with 0x prefix) and will fail to deserialize something like 0x0. Because the specs say this value is Bytes8 https://github.com/ethereum/execution-specs/blob/master/src/ethereum/muir_glacier/eth_types.py#L118 and geth also implements it as such, I think failing to deserialize 0x0 for the nonce is reasonable